### PR TITLE
Replace "java.util.Optional" String with `class.java.name` to prevent…

### PR DESCRIPTION
**Pull request by MohammadEhsanMohit**
When the project uses desugaring to replace `java.util.Optional`, it can lead to exceptions in the Courier library on Android devices that do not support Java 8. To prevent this issue, it is advisable to use `Optional::class.java.name`, as this approach avoids any unintended changes introduced by R8 or desugaring processes. Additionally, since devices running Android versions lower than 7 do not support Java 8, there is no need to implement specific checks for these devices, as they will inherently be excluded from Java 8 features.

### CODE REVIEW

**Comment by sumitjat3** [comment]
@MohammadEhsanMohit  Change LGTM, hoping you've verified the changes and it's fixing the given crash.

Thanks for contribution

File: buildSrc/src/main/kotlin/deps.kt [hunk 1]
```diff
@@ -134,4 +134,4 @@ object deps {
         const val runtime = "androidx.work:work-runtime:${versions.workManager}"
         const val runtime_2_6_0 = "androidx.work:work-runtime:2.6.0"
     }
-}
\ No newline at end of file
+}
```
Comment (sumitjat3) [inline buildSrc/src/main/kotlin/deps.kt:137]: Can you revert these changes

**Review by sumitjat3 — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/kotlin/deps.kt (hunk 1 above)
Comment (MohammadEhsanMohit) [reply buildSrc/src/main/kotlin/deps.kt:137]: Done!

**Review by sumitjat3 — APPROVED** [review]

**Comment by deepanshu42** [comment]
Hi @MohammadEhsanMohit thank you for contribution. Changes look good, will merge and publish a new version of courier android library.

**Review by deepanshu42 — APPROVED** [review]

**Pull request merged**